### ISSUE TITLE
Url Routing

### ISSUE THREAD

**Issue by Gertjv20** [twitchax/AspNetCore.Proxy#39]
Good Day

First I would like to thank you for this project and the dedication to keep supporting it.

I have a problem that might be caused by lack of knowledge from my side because it seems like no one else have this issue. So I do apologize before hand if its something obvious that I have overlooked or not aware of.

Basically what it comes down to is let say I have a controller with with a action and the route is the following.

`[Route("api/google/{**rest}`
`public Task TestProxy(string rest)`
`{`
`      var queryString = this.Request.QueryString.Value;`
`      return this.HttpProxyAsync($"https://www.google.com/{rest}{queryString}"`
`}`

If I go to this route initially it is working as expected I get the following url 
"https://localhost:5001/api/google/"

The logo of google does not display but what I will be pointing out next will be the reason for that.
If I then type in the web page of google test and select to search it immediately tries and send the request directly to "https://localhost:5001/search?source" and not "https://localhost:5001/api/google/search?source" where my action route is. This is the same reason why the image of google is not showing.

As a workaround I have made an action with a catchall route to redirect back to the proxy action. This then does work and even the google page is showing correctly. But I believe it is not intended to work this way. 

The main reason I am using this is to access local copier devices on the local network with their web  interfaces but you need to authenticate with the web interface witch makes use of "cgi" and using the workaround that I mentioned when trying to log in it just doesn't do anything and the browser is waiting for a reply.

Your feedback will be much appreciated.

**Comment by twitchax** [comment]
Hi @Gertjv20,

Thanks for filing the issue and for the interest! :)

This is actually a scenario I have run into before.  Unfortunately, this scenario is not really what this library is designed for.  This library is more designed to "reverse" proxy a server, or something like that.  And, in that vein, it is meant to proxy just one request coming from a client.

The problem that you are running into (using the google example), is that a request to `https://localhost:5001/api/google/` will forward, as you have instructed, to `https://www.google.com/`.  However, the local google page does not know about your proxy, so it has a bunch of references to things like `/search` or `/favicon.ico`.  As far as the browser is concerned, a reference to `/favicon.ico` is a reference to `https://localhost:5001/`, since it doesn't really know about the proxy magic that you are doing.

However, there is a possible solution (though I have not tried it).  You could set a `base` in all HTML (see: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base).  Basically, you would do something like this.

```csharp
private HttpProxyOptions _httpOptions = HttpProxyOptionsBuilder.Instance
        .WithAfterReceive(async (c, hrm) =>
        {
            // Alter the content in  some way before sending back to client.
            var content = await hrm.Content.ReadAsStringAsync();

            // Do something here that (1) detects whether or not this response is HTML, and (2) injects a base tag into the HTML at the proper location.

            hrm.Content = new StringContent(content);
        }).Build();

[Route("api/google/{**rest}
public Task TestProxy(string rest)
{
    var queryString = this.Request.QueryString.Value;
    return this.HttpProxyAsync($"https://www.google.com/{rest}{queryString}");
}
```

**Comment by Gertjv20** [comment]
Hi Twitchax

Thank you for your response. I do understand what you are saying and I will definitely try your suggested solution. From what you are describing is definitely the reason of the workaround I have done which did not work but I am eager to try what you suggested. Unfortunately I will only be able to test it on Monday once I am back at work to see if it will work with the copier web interface that is on that network.

I am actually planning on trying something bazaar. This is only one piece of the puzzle of the project. Just for interest sake I will elaborate what I have in mind. I have never seen this and I was not able to get any example of what I am trying to accomplish.

What I have is a asp net core server and I have a winform/wpf combined application to let it work as a tray app. This client communicates with the copier devices to get their meter readings and then after a set poll time communicates with the Asp server sending that information. I have implemented SignalR to be able to send different poll times to the client db witch helps in not waiting for the next poll time of the client before it sees the change on the server.

So what I have described above I already have in place. What I would like to achieve next is for the server to have a route that sends the copier local IP of the URL I would like to retrieve on the network of the client so that the client would do the web request and send it back to the server in order to view that web response as the web page on the server route. And of course I would like to interact with that web page on the server side that forward those requests again to the server.

I have not seen anything like it before but I am sure there must be a way to achieve this.
So my thought process was I must at least be able to get the server to forward a request and return to view it as a web page. Thus the reason I discovered your project.

For what I am trying to accomplish this might not even be the correct approach.

Thank you for the kind response.

**Comment by Gertjv20** [comment]
Hi Twitchax

Just for feedback I have tried your suggested solution. It unfortunately did not work in a way that would have benefited me. 

I did inject the base tag with the href as my server route and I could see if I look at the source code of the page that it indeed showed it as the new content but still it would not return the remaining requests of the page back to the route specified as base. When I changed the base href to google then it worked and the page showed correctly but then selecting any element on the page would take me to the URL directly which I would expect because of the base href.

Anyways I decided not to implement the feature in my project because on second thought clients might have a problem if we access web interfaces on their local network remotely.

Thank you for your time and effort it is much appreciated.
